Awesome books - First pull request

ReshyResh opened the pull request:
## Project requirements succesfully fulfilled
  - Create a collection that keeps a list of books
  - Create a function to add a new book to the collection, with title and author.
  - Create a function to remove a book from the collection
  - Display all books saved in the collection in the top part of the page.
  - Make sure that data is preserved in the browser's memory by using localStorage.

# Screenshot
![pJ9t1x7](https://user-images.githubusercontent.com/85108160/128715180-7af2e91e-f744-4210-b465-f30b091fd6f0.png)

# Live demo
See the project in action [here](https://reshyresh.github.io/Awesome-Books-JS-Project/)

mimipeshy reviewed (changes requested):
## Hi @ReshyResh and @coldatlas  👋🏾 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
#### Highlights

- I can add and remove books✔️ 
- Good GitHub workflow ✔️ 
- Data is persisted ✔️ 

## Required Changes ♻️
- [x] Kindly enable [github workflows and linter configurations
](https://github.com/microverseinc/linters-config/tree/master/html-css-js)
- [x] Kindly add a README file. You can use [this template](https://github.com/microverseinc/readme-template)
- [x] _Check the comments under the review._


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](<https://microverse.zendesk.com/hc/en-us/articles/1500004088561>) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](<https://airtable.com/shrQAqnBwek5a0O0s>)._

mimipeshy commented on script.js:
```diff
@@ -0,0 +1,83 @@
+  count.id = 'counter';
+  box.appendChild(count);
+}, 1);
+function add() {
+  const form = document.getElementById('books-form');
+  const title = form.title.value;
+  const author = form.author.value;
+  const box = document.getElementById('books');
+  if (title !== '' && author !== '') {
```
Kindly ensure arrow functions are used instead of the function keyword if you check the Javascript requirements you would see that the use of ES6 is encouraged. Javascript's ES6 brings new syntax and awesome features to make your code more modern and readable. It allows you to write less code and do more. ES6 introduces us to many great features like arrow functions. The arrow function is really awesome and makes your code more readable, more structured, and look modern. Take a minute and also go through this article on why [ES6 arrow functions are so awesome.](link -> https://zendev.com/2018/10/01/javascript-arrow-functions-how-why-when.html)

mimipeshy commented on script.js:
```diff
@@ -0,0 +1,83 @@
+  }
+}
+
+function remove(n) {
+  books.splice(n, 1);
+  const toremove = document.getElementById(n);
+  const box = document.getElementById('books');
+  const children = box.getElementsByTagName('div');
+  for (let i = n; i < children.length; i += 1) {
+    const button = children[i].getElementsByTagName('button');
+    button[0].setAttribute('onclick', `remove(${children[i].id - 1}),store()`);
+    children[i].id -= 1;
+  }
+  box.removeChild(toremove);
+  document.getElementById('counter').innerHTML = `Total number of books:${books.length}`;
+  id -= 1;
+}
+
```
- [ ] Kindly use `arrow functions` over `function` keyword

mimipeshy commented on script.js:
```diff
@@ -0,0 +1,83 @@
+  id -= 1;
+}
+
+function store() {
+  const box = document.getElementById('books');
+  const children = box.getElementsByTagName('div');
+  if (children.length === 0) {
+    id = 0;
+  }
+  localStorage.setItem('storage', JSON.stringify(books));
+  localStorage.setItem('storage2', id);
+}
+
```
- [ ] Kindly use `arrow` functions over `function` keyword

mimipeshy commented on script.js:
```diff
@@ -0,0 +1,83 @@
+  localStorage.setItem('storage2', id);
+}
+
+function populate() {
+  if (localStorage.getItem('storage')) {
+    const items = localStorage.getItem('storage');
+    const parsed = JSON.parse(items);
+    for (let i = 0; i < parsed.length; i += 1) {
+      const box = document.getElementById('books');
+      const bookdiv = document.createElement('div');
+      bookdiv.id = i;
+      bookdiv.innerHTML = `<p>${parsed[i].title}</p><p>${parsed[i].author}</p><button type="button" onclick="remove(${i}),store()">Remove</button><hr>`;
+      box.appendChild(bookdiv);
```
- [ ] KIndly use `arrow` functions over `function` keyword

ReshyResh commented:
Hello @mimipeshy ,
Thanks for the input, I managed to go through the article about arrow functions and changed the code accordingly.
I also added a readme file and added linter files.
Thanks for the feedback!

mimipeshy reviewed (approved):
## Hi @ReshyResh and @coldatlas 👋🏾 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
